### ISSUE TITLE
Subqueries

### ISSUE THREAD

**Issue by nkovacs** [go-ozzo/ozzo-dbx#32]
`SelectQuery` doesn't implement `Expression`, so I can't use it in `ExistsExp` or `InExp`.

I wrote a function to convert it into an `Expression`:

```go
func SelectExp(query *dbx.SelectQuery) dbx.Expression {
	built := query.Build()
	return dbx.NewExp(built.SQL(), built.Params())
}
```

Is this a good idea? Is there a better way to do this (I don't want to write the whole subquery using raw sql)?

**Comment by kolkov** [comment]
Good call. `SelectQuery` not implementing `Expression` makes subqueries harder than they should be. Your `SelectExp()` workaround using `Build()` to extract the SQL and params is actually the right approach.

Making `SelectQuery` implement `Expression` natively would be a nice improvement. This has been open a long time — still interested? We'll close in a couple of weeks if no response.
